tests/components/homekit_controller/test_connection.py

```diff
@@ -176,3 +180,111 @@ async def test_migrate_ble_unique_id(hass: HomeAssistant) -> None:
     await hass.async_block_till_done()
 
     assert config_entry.unique_id == "02:03:ef:02:03:ef"
+
+
+async def test_thread_provision_no_creds(hass: HomeAssistant) -> None:
+    """Test that we don't migrate to thread when there are no creds available."""
+    accessories = await setup_accessories_from_file(hass, "anker_eufycam.json")
+
+    fake_controller = await setup_platform(hass)
+    await fake_controller.add_paired_device(accessories, "02:03:EF:02:03:EF")
+    config_entry = MockConfigEntry(
+        version=1,
+        domain="homekit_controller",
+        entry_id="TestData",
+        data={"AccessoryPairingID": "02:03:EF:02:03:EF"},
+        title="test",
+        unique_id="02:03:ef:02:03:ef",
+    )
+    config_entry.add_to_hass(hass)
+
+    fake_controller.transport_type = TransportType.BLE
+
+    await hass.config_entries.async_setup(config_entry.entry_id)
+    await hass.async_block_till_done()
+
+    with pytest.raises(HomeAssistantError):
+        await hass.data[KNOWN_DEVICES]["02:03:EF:02:03:EF"].async_thread_provision()
```

Comment:
We aren't supposed to access `hass.data` in tests.

Can you call the button entity press service instead?